UnityProject/Assets/Scripts/Items/Tool/Pickaxe.cs

```diff
@@ -8,6 +8,11 @@
 public class Pickaxe : MonoBehaviour, ICheckedInteractable<PositionalHandApply>
 {
 	[SerializeField] private AddressableAudioSource pickaxeSound = null;
+
+	[Tooltip("How much does this tool multiply mining time")] [SerializeField]
```

Comment:
Could you put this in the toolutil , so you just add the tool component to the pickaxe and make it call the toolutil function you told with the function you want to activate once it's finished
void ProgressComplete()
		{stuff you want to do when finished}
ServerUseToolWithActionMessages(interaction, default time, text for action, ect. ect. , ProgressComplete )

ServerUseToolWithActionMessages is a good example of the lambda function thing

Reply:
Done, but there's some kind of issue with it that I can't identify.

Reply:
Some kind of issue? Something preventing merge?